gpt/git/tools.js

```diff
@@ -0,0 +1,257 @@
+     }
+   }
+
+   class GitAddTool extends GitTool {
+     static lc_name() {
+       return "GitAddTool";
+     }
+
+     constructor(args) {
+       super(args);
+       this.name = "git_add";
+       this.schema = z.object({
+         files: z.string(),
+       });
+     }
+
+     async _call({ files }) {
+       return this.runGitCommand(`add ${files}`);
+     }
+
+     get description() {
+       return `Add files to staging. Use '.' to add all files.`;
+     }
+   }
+
+   class GitNewBranchTool extends GitTool {
+     static lc_name() {
+       return "GitNewBranchTool";
+     }
+
+     name = "git_new_branch";
+
+     schema = z.object({
+       branchName: z.string(),
+     });
+
+     async _call({ branchName }) {
+       return this.runGitCommand(`checkout -b ${branchName}`);
+     }
+
+     description = `Create a new branch.`;
+   }
+
+   class GitCheckoutBranchTool extends GitTool {
+     static lc_name() {
+       return "GitCheckoutBranchTool";
+     }
+
+     name = "git_checkout_branch";
+
+     schema = z.object({
+       branchName: z.string(),
+     });
+
+     async _call({ branchName }) {
+       return this.runGitCommand(`checkout ${branchName}`);
+     }
+
+     description = `Switch to an existing branch.`;
+   }
+
+   class GitPullTool extends GitTool {
+     static lc_name() {
+       return "GitPullTool";
+     }
+
+     name = "git_pull";
+
+     schema = z.object({});
+
+     async _call() {
+       return this.runGitCommand('pull');
+     }
+
+     description = `Pull latest changes from remote.`;
+   }
+
+   class GitPushTool extends GitTool {
+     static lc_name() {
+       return "GitPushTool";
+     }
+
+     name = "git_push";
+
+     schema = z.object({});
+
+     async _call() {
+       return this.runGitCommand('push');
+     }
+
+     description = `Push local commits to remote.`;
+   }
+
+   class GitBranchListTool extends GitTool {
+     static lc_name() {
+       return "GitBranchListTool";
+     }
+
+     name = "git_branch_list";
+
+     schema = z.object({});
+
+     async _call() {
+       return this.runGitCommand('branch');
+     }
+
+     description = `List all local branches.`;
+   }
+
+   class GitDeleteBranchTool extends GitTool {
+     static lc_name() {
+       return "GitDeleteBranchTool";
+     }
+
+     name = "git_delete_branch";
+
+     schema = z.object({
+       branchName: z.string(),
+     });
+
+     async _call({ branchName }) {
+       return this.runGitCommand(`branch -d ${branchName}`);
+     }
+
+     description = `Delete a local branch.`;
+   }
+
+   module.exports = {
+     GitTool,
+     GitDiffStagedTool,
+     GitStagedFilesTool,
+     GitDiffTool,
+     GitStatusTool,
+     GitCommitTool,
+     GitAddTool,
+     GitNewBranchTool,
+     GitCheckoutBranchTool,
+     GitPullTool,
+     GitPushTool,
+     GitBranchListTool,
+     GitDeleteBranchTool
+   };
+   
\ No newline at end of file
```

Comment:
The GitTool classes have a lot of repetitive code. Consider creating a base class with common functionality and extending it for each specific tool to improve maintainability.